Update equipment

manolobevia opened the pull request:
Hi Nick, please review this PR when you have a moment. Thanks

manolobevia committed:
Add reserve items to equipment page

manolobevia committed:
Made grid responsive

manolobevia committed:
Add accesibility link

manolobevia committed:
Add temporary hack to shame css component

manolobevia committed:
Remove underline links on buttons

cappadona committed:
Create equipment type array & loop through

Went with the singular variant for all.

cappadona committed:
Use children from pages drop to build equip list

cappadona committed:
Dynamically build reserve URLs via Actions API

Introduces two new fields to sensitive data site-metafields due to
authentication requirements for LibCal v1.1 API.

Includes debugging info we should be removed before we merge.

Todo:

* update equipment pages (title, slug & handle should be singular)
  -- discussed in Slack
  -- LibCal categories have already been updated
  -- none of the reserve URLs will work properly until this happens
  -- okay to make changes to pages locally, but don't commit

* remove debugging info

* add logic to handle categories/types that are not reservable

cappadona committed:
Build equipment list & links dynamically via children pages & LibCal API (#921)

* Create equipment type array & loop through

Went with the singular variant for all.

* Use children from pages drop to build equip list

* Dynamically build reserve URLs via Actions API

Introduces two new fields to sensitive data site-metafields due to
authentication requirements for LibCal v1.1 API.

Includes debugging info we should be removed before we merge.

Todo:

* update equipment pages (title, slug & handle should be singular)
  -- discussed in Slack
  -- LibCal categories have already been updated
  -- none of the reserve URLs will work properly until this happens
  -- okay to make changes to pages locally, but don't commit

* remove debugging info

* add logic to handle categories/types that are not reservable

manolobevia committed:
Merge commit '49d682976980569f235702db9d7cff9185bb6b82' into update-equipmet

cappadona committed:
Sync equip pages name changes (singular) from prod

We still have to add a page for the thermal imager.

cappadona committed:
Sync camera page title change from prod

Now `DSLR Camera`.

cappadona committed:
Refactor Libservices equipment type logic (#923)

* Refactor Libservices equipment type logic

Cleanup on aisle 6. Still a mess and still hardwired, but should hold us
over for the interim solution proposed in #886.

* Drop all garbage equipment types while we're at it

Cleanup on aisle 7? More Libservices tomfoolery...

manolobevia committed:
Remove LibCal data debug
Add logic for disabling reserve buttons when equipment is not reservable

cappadona commented on app/views/snippets/equipment-list.liquid:
```diff
@@ -30,9 +30,11 @@
               <a href="{% path_to child %}" title="Pick up your {{ child.title }}">
                 <button class="">Pick up Now</button>
               </a>
+              {% if libcal_url %}
```
Nice...you could use [liquid's `capture`](http://shopify.github.io/liquid/tags/variable/#capture) here to simplify this a bit.

Review [this example from the `form-site-feedback` snippet](https://github.com/cul-it/mann-wagon/blob/474378592edd88b5a790717ab1294165220991e9/app/views/snippets/form-site-feedback.liquid#L4-L10) that includes conditionals within the capture:

```liquid
{% capture page_in_question %}
  {% if post? %}
    {{ site_feedback.url }}
  {% else %}
    {{ referer }}
  {% endif %}
{% endcapture %}
```

manolobevia committed:
Use capture liquid tag for reserving equipment

cappadona committed:
GPS model is Garmin eTrex 30x

cappadona committed:
Drop old loanable equipment list setup snippet

No longer used.

cappadona committed:
Quick tooltip to explain disabled reserve button

Not all equipment is reservable. Would have liked to use semantic-ui css
tooltip but that got tricky with opacity and z-index (side-effect of
disabled attribute).

cappadona committed:
Cleanup policies and reserve helper callout

Simpler and sweeter. Also link to hours.

manolobevia committed:
Fix some sass lint warnings

cappadona committed:
Drop temporary style hiding page content

Definitely don't want this in production even though for the time being
we plan on managing most of the copy within the snippet. But this will
be refactored to use editable-elements if this interim solution sticks
around for a bit.

cappadona committed:
Expand scope of style for underlining links

Target the `.page-content` and `.page-content--dynamic` divs instead of
simply paragraphs. This provides better coverage due to dirty/messy
markup introduced by copy/pasting into back-office wysiwyg and also has
added benefit of including linked headings.

Also move the rule to `_a11y` component, where it should have been from
the beginning when introduced in 117427c as part of #907.
